File: javascript/grains/grains.js
var BigInt = require('./big-integer');

var Grains = function() {};

Grains.prototype.square = function(number) {
  return (new BigInt(2)).pow(number - 1).toString();
};

Grains.prototype.total = function() {
  var total = new BigInt(0);
  for (var i = 0; i < 64; i++)
    total = total.add(new BigInt(this.square(i + 1)));
  return total.toString();
};

module.exports = Grains;

File: go/grains/grains.go
package grains

import (
	"fmt"
	"math"
)

// Square ....
func Square(x int) (uint64, error) {
	if x < 1 || x > 64 {
		return uint64(0), fmt.Errorf("nope")
	}
	return uint64(math.Pow(2, float64(x-1))), nil
}

// Total ...
func Total() uint64 {
	tot := uint64(0)
	for i := 0; i < 64; i++ {
		tot += uint64(math.Pow(2, float64(i)))
	}
	return tot
}
